lib/matplotlib/patheffects.py

```diff
@@ -373,3 +375,168 @@ def draw_path(self, renderer, gc, tpath, affine, rgbFace):
+            The offset to apply to the path, in points.
+        spacing : float
+            The spacing between ticks in points.
+            Default is 10.0.
+        angle : float
+            The angle between the path and the tick in degrees.  The angle
+            is measured as if you were an ant walking along the curve, with
+            zero degrees pointing directly ahead, 90 to your left, 180
+            behind you, and 270 to your right.
+            Default is 45.0.
+        length : float
+            The length of the tick relative to spacing.
+            Recommend length=1.0 when angle=90 and length=2.0 when angle=60.
+            Default is 1.414 (sqrt(2)) best for angle=45.
+        flip : bool
+            Flag to flip the ticks to the other side of the line.
+            Default is False.
+        **kwargs
+            Extra keywords are stored and passed through to
+            :meth:`AbstractPathEffect._update_gc`.
+
+        For examples see
+            :doc:`/gallery/misc/tickedstroke_demo`.
+        """
+        super().__init__(offset)
+
+        self._spacing = spacing
+        self._angle = angle
+        self._length = length
+        self._flip = flip
+        self._gc = kwargs
+
+    def draw_path(self, renderer, gc, tpath, affine, rgbFace):
+        """
+        Draw the path with updated gc.
+        """
+        # Do not modify the input! Use copy instead.
+        gc0 = renderer.new_gc()
+        gc0.copy_properties(gc)
+
+        gc0 = self._update_gc(gc0, self._gc)
+        trans = affine + self._offset_transform(renderer)
+
+        thetar = -np.radians(self._angle)
+        trans_matrix = np.array([[np.cos(thetar), -np.sin(thetar)],
+                                 [np.sin(thetar), np.cos(thetar)]])
+
+        if self._flip:
+            iflip = -1
+        else:
+            iflip = 1
+
+        # Convert spacing parameter to pixels.
+        spcpx = renderer.points_to_pixels(self._spacing)
+
+        # Transform before evaluation because to_polygons works at resolution
+        # of one -- assuming it is working in pixel space.
+        transpath = affine.transform_path(tpath)
+
+        # Evaluate path to straight line segments that can be used to
+        # construct line ticks.
+        polys = transpath.to_polygons(closed_only=False)
+
+        for p in polys:
+            x = p[:, 0]
+            y = p[:, 1]
+
+            # Can not interpolate points or draw line if only one point in
+            # polyline.
+            if x.size < 2:
+                continue
+
+            # Find distance between points on the line
+            ds = np.hypot(x[1:] - x[:-1], + y[1:] - y[:-1])
+
+            # Build parametric coordinate along curve
+            s = np.concatenate(([0.0], np.cumsum(ds)))
+            stot = s[-1]
+
+            num = int(np.ceil(stot / spcpx))-1
+            # Pick parameter values for ticks.
+            s_tick = np.linspace(spcpx/2, stot-spcpx/2, num)
```

Comment:
Is this different from?
```suggestion
            # Pick parameter values for ticks.
            s_tick = np.arange(spcpx/2, stot, spcpx)
```

Reply:
I think they work out to the same answer.  I believe the linspace solution is more clear because it makes the start/end values of s_tick very clear.  The arange solution does not make the end point clear at first read.

The explicit calculation of num also makes it very clear that there is one less tick than would result from stot/spcpx.  The arange based code leaves num as a fallout.

An earlier rev of this code used linspace from 0 to stot.  It did not need the -1 to calculate num.  Shifting the ticks to start on the half-spacing improves the look of the lines near plot bounds and in the legend.

Reply:
A perhaps more significant difference.  The arange approach will result in exact spacing between ticks (of spcpx).  it will start at spcpx/2 and will end with a gap between zero and spcpx.

The linspace approach results in exactly spcpx/2 spacing at both ends -- with approximate spcpx spacing along the line.  The error in how many spcpx fit is spread evenly along the line -- vs. all the error showing up at the end of the line with the arange approach.

Because of this, the linspace approach will look more consistent.

Particularly for closed curves (like the circle example), with the linspace approach, the start/end of the curve is not visible - whereas the arange approach will result in un-even spacing at the start/end of the curve.